### PR TITLE
Exposed toolbar config as editor prop...

### LINKED ISSUE #72: Ability to configure toolbar
The availability, order and grouping of toolbar items should be customizable via prop(s).

### PR DESCRIPTION
**Pull request by skolmer**
…and added className prop to toolbar items. Fixes #72

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -59,6 +59,41 @@ class MyStatefulEditor extends Component {
...
+
+```javascript
+
+render () {
+  // The toolbarConfig object allows you to specify custom buttons, reorder buttons and to add custom css classes.
+  // Supported inline styles: https://github.com/facebook/draft-js/blob/master/docs/Advanced-Topics-Inline-Styles.md
+  // Supported block types: https://github.com/facebook/draft-js/blob/master/docs/Advanced-Topics-Custom-Block-Render.md#draft-default-block-render-map
+  const toolbarConfig = {
+    INLINE_STYLE_BUTTONS: [
+      {label: 'Bold', style: 'BOLD', className: 'custom-css-class},
```
Comment (sstur) [inline README.md]: Oops, missing closing quote

File: README.md [hunk 2]
```diff
@@ -59,6 +59,41 @@ class MyStatefulEditor extends Component {
 }
 ```
 
+
+### Toolbar Customization
+
+```javascript
+
+render () {
```
Comment (sstur) [inline README.md]: Nit: remove space after method name.

File: src/RichTextEditor.js [hunk 3]
```diff
@@ -45,6 +46,7 @@ type Props = {
   handleReturn?: (event: Object) => boolean;
   readOnly?: boolean;
   disabled?: boolean; // Alias of readOnly
+  toolbarConfig?: EditorToolbarConfig;
```
Comment (sstur) [inline src/RichTextEditor.js]: I believe this line is incorrect. Th right hand side must be a type annotation, not an object (although the two looks similar at times). I'll try to come up with an example of how to fix it.

File: src/lib/EditorToolbar.js [hunk 4]
```diff
@@ -32,6 +28,7 @@ type Props = {
   keyEmitter: EventEmitter;
   onChange: ChangeHandler;
   focusEditor: Function;
+  toolbarConfig: EditorToolbarConfig;
```
Comment (sstur) [inline src/lib/EditorToolbar.js]: .. same as above.

File: src/lib/EditorToolbar.js [hunk 5]
```diff
@@ -61,22 +58,22 @@ export default class EditorToolbar extends Component {
...
+    const {className, toolbarConfig} = this.props;
     return (
       <div className={cx(styles.root, className)}>
-        {this._renderInlineStyleButtons()}
-        {this._renderBlockTypeButtons()}
+        {this._renderInlineStyleButtons(toolbarConfig || EditorToolbarConfig)}
+        {this._renderBlockTypeButtons(toolbarConfig || EditorToolbarConfig)}
         {this._renderLinkButtons()}
-        {this._renderBlockTypeDropdown()}
+        {this._renderBlockTypeDropdown(toolbarConfig || EditorToolbarConfig)}
```
Comment (sstur) [inline src/lib/EditorToolbar.js]: I realize now we should call this `DefaultEditorToolbarConfig`, but that's not too important.

File: src/lib/EditorToolbar.js [hunk 6]
```diff
@@ -61,22 +58,22 @@ export default class EditorToolbar extends Component {
...
         {this._renderUndoRedo()}
       </div>
     );
   }
 
-  _renderBlockTypeDropdown() {
+  _renderBlockTypeDropdown(toolbarConfig: EditorToolbarConfig) {
```
Comment (sstur) [inline src/lib/EditorToolbar.js]: ..I think this type annotation is wrong also.

File: src/lib/EditorToolbar.js [hunk 7]
```diff
@@ -92,32 +89,34 @@ export default class EditorToolbar extends Component {
     );
   }
 
-  _renderBlockTypeButtons() {
+  _renderBlockTypeButtons(toolbarConfig: EditorToolbarConfig) {
```
Comment (sstur) [inline src/lib/EditorToolbar.js]: .. same.

File: src/lib/EditorToolbar.js [hunk 8]
```diff
@@ -92,32 +89,34 @@ export default class EditorToolbar extends Component {
...
+        className={type.className}
       />
     ));
     return (
       <ButtonGroup>{buttons}</ButtonGroup>
     );
   }
 
-  _renderInlineStyleButtons() {
+  _renderInlineStyleButtons(toolbarConfig: EditorToolbarConfig) {
```
Comment (sstur) [inline src/lib/EditorToolbar.js]: .. same.

File: src/ui/Dropdown.js [hunk 9]
```diff
@@ -6,7 +6,7 @@ import cx from 'classnames';
 import styles from './Dropdown.css';
 
 type Props = {
-  choices: Map<string, string>;
+  choices: Map<string, {label:string, className?:string}>;
```
Comment (sstur) [inline src/ui/Dropdown.js]: Nit: space after `:`

File: src/ui/Dropdown.js [hunk 10]
```diff
@@ -24,6 +24,7 @@ export default class Dropdown extends Component {
     let {choices, selectedKey, className, ...otherProps} = this.props;
     className = cx(className, styles.root);
     let selectedValue = (selectedKey == null) ? '' : choices.get(selectedKey);
+    selectedValue = (selectedValue == null) ? '' : selectedValue.label || '';
```
Comment (sstur) [inline src/ui/Dropdown.js]: For clarity, let's call these two things a different name. Something like:
```js
let selectedItem = (selectedKey == null) ? null : choices.get(selectedKey);
let selectedValue = selectedItem && selectedItem.label || '';
```

File: src/RichTextEditor.js (hunk 3 above)
Comment (skolmer) [inline src/RichTextEditor.js]: An example would be indeed helpful. I'm not used to flow type annotations. Thanks for the great code review!

**Comment by sstur** [comment]
I was trying to figure out how Travis passed with those Flowtype issues (the object where a type annotation should go) and I just realized it's because I forgot a `@flow` directive at the top of `src/lib/EditorToolbarConfig.js`

If you add that, then Flow will show the error:
```
object literal: ...
Ineligible value used in/as type annotation
```

**Comment by sstur** [comment]
I've fixed the issues mentioned in [this commit](https://github.com/sstur/react-rte/pull/107/commits/ec3e303135d0192463e19ffb7dd88e2f7ea54927)

Just waiting on TravisCI and we can land this one.

Thanks!

**Pull request merged**